src/metpy/calc/thermo.py

```diff
@@ -973,12 +971,6 @@ def _parcel_profile_helper(pressure, temperature, dewpoint,
     temp_upper = moist_lapse(unique[::-1], temp_lower[-1]).to(temp_lower.units)
```

Comment:
Similarly, the temp_upper should be computed as before, *before* doing any of the virtual temp corrections. Wait until after computing the parcel profile using the original temp, dewpoint, etc., and *then* calculated the virtual temperature of the entire path.

Reply:
In other words, move the code blocks that are computing the virtual temperature for below and above the LCL to the end of this section after everything else.